index.js

```diff
@@ -46,6 +90,12 @@ module.exports = function (source, map) {
     config = Object.assign({ }, config, resourceConfig);
   }
 
+  Object.entries(config).forEach(([key, value]) => {
```

Comment:
Rather than casting the provided config based on a whitelist, why not look up what type the config should have in _sharp_, since it is initialized in its constructor [here](https://github.com/lovell/sharp/blob/master/lib/constructor.js#L125)? Like (this should be moved after line 100):

```js
Object.entries(config).forEach(([key, value]) => {
  if (typeof transformer[key] === 'number') {
    config[key] = Number(value);
  }
});
```